### ISSUE TITLE
bug: ion-alert traps keyboard focus

### ISSUE THREAD

**Issue by kensodemann** [ionic-team/ionic-framework#21647]
# Bug Report

**Ionic version:**

[ ] **4.x**
[x] **5.x**

**Current behavior:**
ion-alert does not trap the keyboard focus within the subsections of content on a web page. Instead, when you get to the last element in the alert (such as the "OK" button), hitting tab takes you up in into the web browser controls rather than the first element in the modal. See the Material Design example linked below for a modal that behaves properly, not taking the user back to the browser controls.

This is violating the guideline on accessibility

Please refer to the WCAG guideline: https://www.w3.org/TR/UNDERSTANDING-WCAG20/keyboard-operation-trapping.html. Refer to the success criterion about the Modal box mentioned in above link.

Issue is reproducible from the ionic website link: https://ionicframework.com/docs/demos/api/alert/index.html?ionic:mode=ios

Sample implementation which meets the success criteria for reference: https://material.angular.io/components/dialog/overview

**Comment by kensodemann** [comment]
Here is a recreation repo using @ionic/anguar@5.2.2:
https://github.com/mahiii85/ionic-test.git ; branch: feature/ion-alert-test

**Comment by liamdebeasi** [comment]
Resolved via https://github.com/ionic-team/ionic-framework/pull/21716
